src/coreComponents/constitutive/fluid/PVTFunctions/BrineCO2DensityFunction.cpp

```diff
@@ -18,77 +18,62 @@
 
 #include "constitutive/fluid/PVTFunctions/BrineCO2DensityFunction.hpp"
 
+#include "constitutive/fluid/PVTFunctions/PVTFunctionHelpers.hpp"
+#include "managers/Functions/FunctionManager.hpp"
+#include "managers/GeosxState.hpp"
+
 namespace geosx
 {
 
 using namespace stringutilities;
 
-namespace PVTProps
+namespace constitutive
 {
 
-BrineCO2DensityFunction::BrineCO2DensityFunction( string_array const & inputPara,
-                                                  string_array const & componentNames,
-                                                  real64_array const & componentMolarWeight ):
-  PVTFunction( inputPara[1], componentNames, componentMolarWeight )
+namespace PVTProps
 {
-  bool notFound = 1;
-
-  for( localIndex i = 0; i < componentNames.size(); ++i )
-  {
-
-    if( componentNames[i] == "CO2" || componentNames[i] == "co2" )
-    {
-      m_CO2Index = i;
-      notFound = 0;
-      break;
-    }
-
-  }
-
-  GEOSX_ERROR_IF( notFound, "Component CO2 is not found!" );
-
-  notFound = 1;
-
-  for( localIndex i = 0; i < componentNames.size(); ++i )
-  {
-
-    if( componentNames[i] =="Water" || componentNames[i] == "water" )
-    {
-      m_waterIndex = i;
-      notFound = 0;
-      break;
-    }
-
-  }
-
-  GEOSX_ERROR_IF( notFound, "Component Water/Brine is not found!" );
 
+BrineCO2Density::BrineCO2Density( array1d< string > const & inputPara,
+                                  array1d< string > const & componentNames,
+                                  array1d< real64 > const & componentMolarWeight ):
+  PVTFunctionBase( inputPara[1],
+                   componentNames,
+                   componentMolarWeight )
+{
+  array1d< string > expectedCO2ComponentNames;
+  expectedCO2ComponentNames.resize( 2 );
+  expectedCO2ComponentNames[0] = "CO2";
+  expectedCO2ComponentNames[1] = "co2";
+  bool found = PVTFunctionHelpers::findName( componentNames, expectedCO2ComponentNames, m_CO2Index );
+  GEOSX_ERROR_IF( !found, "Component CO2 is not found!" );
```

Comment:
If you accept my other suggestion, this can become
```suggestion
  char const * expectedCO2ComponentNames[] = { "CO2", "co2" };
  m_CO2Index = PVTFunctionHelpers::findName( componentNames, expectedCO2ComponentNames );
  GEOSX_ERROR_IF( 0 <= m_CO2Index && m_CO2Index < componentNames.size(), "Component CO2 is not found!" );
```
(might need to deal with signed/unsigned comparison here)

Reply:
Yes, here I implemented your two suggestions, thanks